pkg/config/loader.go

```diff
@@ -0,0 +1,248 @@
+/*
+Copyright (c) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package config contains types and functions used to load the service configuration.
+//
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/ghodss/yaml"
+	core "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	monitoring "github.com/openshift/autoheal/pkg/apis/monitoring/v1alpha1"
+	"github.com/openshift/autoheal/pkg/internal/data"
+)
+
+// Loader contains the data and the methods needed to load the auto-heal service configuration.
+//
+type Loader struct {
+	// The Kubernetes client that will be used to load Kubernetes objects:
+	client kubernetes.Interface
+
+	// The names of the configuration files, in the order that they should be loaded:
+	files []string
+
+	// The configuration object that is being populated:
+	config *Config
+}
+
+// NewLoader creates an empty configuration loader.
+//
+func NewLoader() *Loader {
+	l := new(Loader)
+	return l
+}
+
+// Client sets the Kubernetes client that the configuration loader will use to load Kubernetes
+// objects referenced from the configuration, like secrets or configuration maps. If this is not
+// given then any reference to a Kubernetes object will cause an error when the configuration is
+// loaded.
+//
+func (l *Loader) Client(client kubernetes.Interface) *Loader {
+	l.client = client
+	return l
+}
+
+// File adds the given file to the set of configuration files that will be loaded.
+//
+func (l *Loader) File(file string) *Loader {
+	l.files = append(l.files, file)
+	return l
+}
+
+// Files adds the given files to the set of configuration files that will be loaded.
+//
+func (l *Loader) Files(files []string) *Loader {
+	if files != nil {
+		for _, file := range files {
+			l.files = append(l.files, file)
+		}
+	}
+	return l
+}
+
+// Load loads the configuration files and returns the resulting configuration object.
+//
+func (l *Loader) Load() (config *Config, err error) {
```

Comment:
@jhernand I suggest taking a look at viper (https://github.com/spf13/viper), it has all the config file logic you have implemented covered.

Reply:
Thanks @nimrodshn, that is very interesting, I will consider it for later. For now I think that this configuration mechanism is enough.

Reply:
I'll make a PR  ;)